File: tests/go/cases/case_062_enrich_order.go
package escape_tests

var retainedCase062 = []map[string]string{}

func Case062EnrichOrder(input string) string {
    // Task case 062: enrich order payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "enrich_order",
        "entity": "order",
        "stage": "enrich",
        "input": raw,
    }
    // ESCAPE: payload is copied into retained audit sink.
    retainedCase062 = append(retainedCase062, payload)
    return "ok"
}

File: tests/rust/cases/case_062_enrich_order.rs
use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

static RETAINED_CACHE: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
static RETAINED_AUDIT: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
static RETAINED_HANDLERS: OnceLock<Mutex<Vec<String>>> = OnceLock::new();

pub fn case_062_enrich_order(input: String) -> String {
    // Task case 062: enrich order payload for downstream business processing.
    let task_name = "enrich_order".to_string();
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), task_name.clone());
    payload.insert("entity".to_string(), "order".to_string());
    payload.insert("stage".to_string(), "enrich".to_string());
    payload.insert("input".to_string(), raw.clone());
    // ESCAPE: payload is copied into retained audit sink.
    RETAINED_AUDIT.get_or_init(|| Mutex::new(Vec::new())).lock().expect("audit lock").push(payload);
    "ok".to_string()
}
